### DISCUSSION TITLE
License for OSS library

### DISCUSSION THREAD

**Discussion by jodydonetti** [ServiceStack/Discuss#48] [Q&A] [answered]
## Context

Hi,
I have an OSS library where the serialization aspect is based on a common interface and different implementations can be provided, pretty normal stuff to be flexible and avoid a lock-in for my users. Among others, I'd like to add support ServiceStack's JSON serializer.
The implementation is pretty simple and I've already made it, but before proceeding I had some doubts about the licensing aspect and wanted to double check it.

## RTFM: what I understood

Based on what I've read [here](https://servicestack.net/free) it seems that I would need to generate a license for OSS projects (that should be renewd, etc).

But then by reading [here](https://docs.servicestack.net/register) it seems that a license is needed **only** by actual apps or similar:

> all options only need to be added to your top-level host projects

Now, since my project is a library (and not an app or app-like, so not "top-level") and it will be used by someone else but not on its own, I think I should not have to do anything in particular.
My users instead, when they will eventually use my library in combination with ServiceStack's JSON serializer, will need to create a free license or buy one.

Is this correct?

Also I was wondering If it will be ok for my "bridge" implementation (the little project that implements the common interface and forwards calls to ServiceStack's JSON serializer) to be released as MIT like all the other parts of my library.

I thought about adding some links here for context, but also wanted to avoid seemingly self-promoting my lib.

## So Anyway

So, to sum it all up:
1) should I have to create a license?
2) can I use MIT as the license for my small "bridge" project?

Thanks!

**Comment by mythz** [comment, marked as answer]
You don't need a license to use [ServiceStack.Text which is free to use](https://servicestack.net/text), as such has no technical restrictions that requires a license to unlock so your project wont need to create or register any License to use ServiceStack Text Serializers.

**Reply by jodydonetti** [reply]
Ah, I've missed that page, sorry.
Thanks for the quick response!
